src/event-transformer/index.ts

```diff
@@ -124,19 +122,12 @@ async function run() {
           const results = (await Promise.all(
             messages
               .map((message: any) => JSON.parse(message.value!.toString()))
-              .filter(item => !item.skipped)
-              .map(async (value: any) => {
-                const s3Resp = (await s3.getObject({
-                  Key: value.Key,
-                  Bucket: value.Bucket
-                }).promise())
-                const block: BlockResponse & { slot: number } = JSON.parse(s3Resp.Body!.toString());
-                const ret: KafkaMessage[] = block.transactions
-                  .filter(txn => !txn.meta?.err)
-                  .flatMap((txn: BlockTransaction) => processTxn(transformers, block, txn))
-  
-                return ret;
-              })
+              .map(txn => ({
+                ...txn,
+                transaction: Transaction.from(txn.transaction)
+              }))
+              .filter((txn: ConfirmedTransaction) => !txn.meta?.err)
+              .flatMap((txn: ConfirmedTransaction) => processTxn(transformers, txn))
```

Comment:
This logic may be a bottle neck? At any given time how large could the `messages` array be?